ethdb/pebble/pebble.go

```diff
@@ -308,10 +310,10 @@ func (d *Database) Get(key []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	ret := make([]byte, len(dat))
-	copy(ret, dat)
-	closer.Close()
-	return ret, nil
+	if err = closer.Close(); err != nil {
+		return nil, err
+	}
+	return common.CopyBytes(dat), nil
```

Comment:
From the docs on `db.Get` in pebble/db.go:
```
// ... The returned
// slice will remain valid until the returned Closer is closed. On success, the
// caller MUST call closer.Close() or a memory leak will occur.
```
You changed the order here, and call `Close` before we're finished operating on the returned slice. This PR thus subtly breaks the database api. 

I don't really see any point in this PR. If there's an error in the db, which surfaces on Close here, then it's probably pretty bad (someone yanked the disk?) and it'll wreak havoc somewhere else anyway

Reply:
OTOH, sure, I can agree that it's probably "The Right Thing To Do", to handle errors where they appear. But please make no other changes. Using `common.CopyBytes` here is not great, IMO. Keep the `copy` as is, rather than hope for `common.CopyBytes` to become inlined magically.

Reply:
Great catch, copy operation should done before `Close`. Will be reverted.